pyproject.toml

```diff
@@ -83,7 +83,6 @@ expected_to_fail = [
     "scripts/examples/python/exportimport.py",
     # Other fixes needed:
     "scripts/examples/python/mini-xenrt.py",
-    "scripts/examples/python/XenAPI/XenAPI.py",
```

Comment:
I'm glad if you fix some of the issues, but some need to drop python2 support which is more involved than we want here. I don't recommend dropping this line just yet.

Reply:
Without dropping that line we can't merge this PR, because modifying a python file requires it to be removed from there. I'll take a look what I can fix while keeping it py2 compatible.
We do need to fix the py3 bugs, because SM is already using this in py3, and may likely run into those py3 bugs

Reply:
ah, that's really annoying

Reply:
I added the fixes here, they are not as bad as I thought.

The bool vs int can be worked around based on Python version.

And the other 2 (missing `errno` in `socket`module) seems to actually be a bug in `pytype`, because `socket.errno` is definitely there in python3 too, there is no bug. So I added a comment to suppress it.